src/track/playcounter.cpp

```diff
@@ -1,7 +1,20 @@
 #include "track/playcounter.h"
 
+PlayCounter::PlayCounter(int timesPlayed)
+        : m_iTimesPlayed(timesPlayed),
+          m_bPlayed(false) {
+    m_pPracticemodeEnabled = std::make_shared<ControlProxy>("[Library]", "practice_mode");
```

Comment:
Is this pointer actually shared? 
Could we use a unique_ptr?

Reply:
yes it is, The playCounter is copied/assigned at other points in the codebase, you will get an compile error, if you are using a unique_ptr at this place.